csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -380,5 +380,40 @@ public ReadOnlySpan<byte> GetBytes(int index, out bool isNull)
         }
 
         IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable<byte[]>)this).GetEnumerator();
+
+        int ICollection<byte[]>.Count => Length;
+        bool ICollection<byte[]>.IsReadOnly => true;
+        void ICollection<byte[]>.Add(byte[]? item) => throw new NotSupportedException("Collection is read-only.");
+        bool ICollection<byte[]>.Remove(byte[]? item) => throw new NotSupportedException("Collection is read-only.");
+        void ICollection<byte[]>.Clear() => throw new NotSupportedException("Collection is read-only.");
+
+        bool ICollection<byte[]>.Contains(byte[] item)
+        {
+            for (int index = 0; index < Length; index++)
+            {
+                var foo = GetBytes(index);
+                if(IsSpanDataEqual(foo, item))
+                //if (foo == item)
```

Comment:
remove?

Reply:
I will also rename the variable `foo`. You can tell I was experimenting here, and forgot to finish off this code properly.